Mathlib/Data/Set/Pointwise/BigOperators.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+
+! This file was ported from Lean 3 source module data.set.pointwise.big_operators
+! leanprover-community/mathlib commit 008205aa645b3f194c1da47025c5f110c8406eab
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.BigOperators.Basic
+import Mathlib.Data.Set.Pointwise.Basic
+
+/-!
+# Results about pointwise operations on sets and big operators.
+-/
+
+
+namespace Set
+
+section BigOperators
+
+-- Porting note: commented out the next line
+-- open BigOperators
+
+open Pointwise Function
+
+variable {α : Type _} {ι : Type _} [CommMonoid α]
+
+/-- The n-ary version of `Set.mem_mul`. -/
+@[to_additive " The n-ary version of `Set.mem_add`. "]
+theorem mem_finset_prod (t : Finset ι) (f : ι → Set α) (a : α) :
+    (a ∈ ∏ i in t, f i) ↔ ∃ (g : ι → α)(_ : ∀ {i}, i ∈ t → g i ∈ f i), (∏ i in t, g i) = a := by
+  classical
+    induction' t using Finset.induction_on with i is hi ih generalizing a
+    · simp_rw [Finset.prod_empty, Set.mem_one]
+      exact ⟨fun h ↦ ⟨fun _ ↦ a, fun hi ↦ False.elim (Finset.not_mem_empty _ hi), h.symm⟩,
+        fun ⟨_, _, hf⟩ ↦ hf.symm⟩
+    rw [Finset.prod_insert hi, Set.mem_mul]
+    simp_rw [Finset.prod_insert hi]
+    simp_rw [ih]
+    constructor
+    · rintro ⟨x, y, hx, ⟨g, hg, rfl⟩, rfl⟩
+      refine ⟨Function.update g i x, ?_, ?_⟩
+      · intro j hj
+        obtain rfl | hj := Finset.mem_insert.mp hj
+        · rwa [Function.update_same]
+        · rw [update_noteq (ne_of_mem_of_not_mem hj hi)]
+          exact hg hj
+      · rw [Finset.prod_update_of_not_mem hi, Function.update_same]
+    · rintro ⟨g, hg, rfl⟩
+      exact ⟨g i, is.prod g, hg (is.mem_insert_self _),
+        ⟨⟨g, fun hi ↦ hg (Finset.mem_insert_of_mem hi), rfl⟩, rfl⟩⟩
+#align set.mem_finset_prod Set.mem_finset_prod
```

Comment:
Corresponding align tags have to be added for the theorems generated by `to_additive`. (If you're not sure what they are, you can replace `to_additive` by `to_additive?` and see what it outputs.)

Reply:
Thanks for the review. 

I have added the `align` for the lemmas generated by `to_additive`. I did not do that in my previous PR. So I wonder why is it necessary to do it for this one? Is it because they cannot be auto-generated for some reason?

Reply:
It was also necessary in previous PRs, it's just very easy to overlook :-) There are more details in the porting wiki, there's a whole section about `to_additive`.